src/shims/unix/freebsd/sync.rs

```diff
@@ -0,0 +1,81 @@
+use crate::helpers::EvalContextExt as HEvalContext;
+use crate::sync::EvalContextExt as SEvalContext;
+use crate::thread::EvalContextExt as TEvalContext;
+use crate::InterpResult;
+use crate::OpTy;
+use crate::{MiriEvalContext, Tag};
+use std::fmt::Pointer;
+
+enum OpType {
+    UmtxOpWait = 2,
+    //TODO: other types
+}
+
+impl TryFrom<i32> for OpType {
+    type Error = &'static str;
+
+    fn try_from(value: i32) -> Result<Self, Self::Error> {
+        match value {
+            2 => Ok(OpType::UmtxOpWait),
+            _ => Err("Unsupported Futex operation"),
+        }
+    }
+}
+
+struct Futex<'mir, 'tcx> {}
+
+impl<'mir, 'tcx: 'mir> Futex<'mir, 'tcx> {
+    /// Function that performs a specific action on a futex thread
+    /// https://www.freebsd.org/cgi/man.cgi?query=_umtx_op&sektion=2&n=1
+    ///
+    /// # Arguments
+    ///
+    /// * `this`: Context of the evaluation
+    /// * `obj`: Pointer to a variable of type long
+    /// * `op`: Futex operation to perform on `obj`
+    /// * `val`: Depends on the operation performed (see man pages)
+    /// * `uaddr`: Depends on the operation performed (see man pages)
+    /// * `uaddr2`: Depends on the operation performed (see man pages)
+    ///
+    /// returns: InterpResult<'tcx>
+    pub fn futex<'tcx>(
+        this: &mut MiriEvalContext<'_, 'tcx>,
+        // Object to operate on
+        obj: &OpTy<'tcx, Tag>,
```

Comment:
All the other arguments are already interpreted at some Rust type; please do the same with this one. The comment seems to indicate it's a pointer, so then this should have type `Pointer<Option<Tag>>`.